From 0e899bd233c926f894aa41fb3aa9590a0bd4972b Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Mon, 2 Nov 2015 17:49:54 +0800
Subject: [PATCH 603/635] MA-7185 Align imx HAL folder build config

HAL build should depend on the specific HAL build flag but not soc
Use 2D build flag to build the gralloc/hwc in mx7 folder
Remove non-used HWC for mx5

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 Android.mk                        |  33 ++--------
 hwcomposer/Android.mk             |  29 ---------
 hwcomposer/README.android         |   3 -
 hwcomposer/hwcomposer.cpp         | 132 --------------------------------------
 mx6/hwcomposer/Android.mk         |   2 +
 mx6/libcamera/Android.mk          |   3 +-
 mx6/libcamera2/Android.mk         |   2 +
 mx6/libgralloc_wrapper/Android.mk |   2 +
 mx7/gralloc/Android.mk            |   2 +
 mx7/hwcomposer/Android.mk         |   2 +
 10 files changed, 17 insertions(+), 193 deletions(-)
 delete mode 100644 hwcomposer/Android.mk
 delete mode 100644 hwcomposer/README.android
 delete mode 100644 hwcomposer/hwcomposer.cpp

diff --git a/Android.mk b/Android.mk
index 62a7a16..cadc083 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,29 +1,6 @@
-common_imx_dirs := libsensors libgps lights wlan libbt-ath3k
-mx5x_dirs := $(common_imx_dirs) mx5x/audio mx5x/libcopybit mx5x/libgralloc  mx5x/hwcomposer mx5x/libcamera mx5x/power
-mx6_dirs := $(common_imx_dirs) alsa mx6/libgralloc_wrapper mx6/hwcomposer mx6/power mx6/consumerir
-mx7_dirs := $(common_imx_dirs) alsa mx6/power mx7/gralloc mx7/hwcomposer
+imx_dirs := libsensors libgps lights wlan libbt-ath3k \
+            alsa mx6/libgralloc_wrapper mx6/hwcomposer \
+            mx6/power mx6/consumerir mx7/gralloc mx7/hwcomposer \
+
+include $(call all-named-subdir-makefiles,$(imx_dirs))
 
-ifeq ($(TARGET_BOARD_PLATFORM),imx6)
-  ifeq ($(IMX_CAMERA_HAL_V2),true)
-    mx6_dirs += mx6/libcamera2
-  else
-    mx6_dirs += mx6/libcamera
-  endif
-  include $(call all-named-subdir-makefiles,$(mx6_dirs))
-else
-  ifeq ($(TARGET_BOARD_PLATFORM),imx7)
-    ifeq ($(IMX_CAMERA_HAL_V2),true)
-      mx7_dirs += mx6/libcamera2
-    else
-      mx7_dirs += mx6/libcamera
-    endif
-	ifeq ($(BOARD_USE_SENSOR_FUSION),true)
-		mx7_dirs += libsensors_sensorhub
-	endif
-    include $(call all-named-subdir-makefiles,$(mx7_dirs))
-  else
-    ifeq ($(TARGET_BOARD_PLATFORM),imx5x)
-      include $(call all-named-subdir-makefiles,$(mx5x_dirs))
-    endif
-  endif
-endif
diff --git a/hwcomposer/Android.mk b/hwcomposer/Android.mk
deleted file mode 100644
index ae5b6f3..0000000
--- a/hwcomposer/Android.mk
+++ /dev/null
@@ -1,29 +0,0 @@
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-
-LOCAL_PATH := $(call my-dir)
-
-# HAL module implemenation, not prelinked and stored in
-# hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
-include $(CLEAR_VARS)
-LOCAL_PRELINK_MODULE := false
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
-LOCAL_SHARED_LIBRARIES := liblog libEGL
-LOCAL_SRC_FILES := hwcomposer.cpp
-LOCAL_MODULE := hwcomposer.$(TARGET_BOARD_PLATFORM)
-LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
-LOCAL_MODULE_TAGS := eng
-include $(BUILD_SHARED_LIBRARY)
-
diff --git a/hwcomposer/README.android b/hwcomposer/README.android
deleted file mode 100644
index 4aa7203..0000000
--- a/hwcomposer/README.android
+++ /dev/null
@@ -1,3 +0,0 @@
-
-Skeleton for the "hwcomposer" HAL module.
-
diff --git a/hwcomposer/hwcomposer.cpp b/hwcomposer/hwcomposer.cpp
deleted file mode 100644
index 9e73e47..0000000
--- a/hwcomposer/hwcomposer.cpp
+++ /dev/null
@@ -1,132 +0,0 @@
-/*
- * Copyright (C) 2010 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-#include <hardware/hardware.h>
-#include <hardware/overlay.h>
-
-#include <fcntl.h>
-#include <errno.h>
-
-#include <cutils/log.h>
-#include <cutils/atomic.h>
-
-#include <hardware/hwcomposer.h>
-
-#include <EGL/egl.h>
-
-/*****************************************************************************/
-
-struct hwc_context_t {
-    hwc_composer_device_t device;
-    /* our private state goes below here */
-};
-
-static int hwc_device_open(const struct hw_module_t* module, const char* name,
-        struct hw_device_t** device);
-
-static struct hw_module_methods_t hwc_module_methods = {
-    open: hwc_device_open
-};
-
-hwc_module_t HAL_MODULE_INFO_SYM = {
-    common: {
-        tag: HARDWARE_MODULE_TAG,
-        version_major: 1,
-        version_minor: 0,
-        id: HWC_HARDWARE_MODULE_ID,
-        name: "Sample hwcomposer module",
-        author: "The Android Open Source Project",
-        methods: &hwc_module_methods,
-    }
-};
-
-/*****************************************************************************/
-
-static void dump_layer(hwc_layer_t const* l) {
-    ALOGD("\ttype=%d, flags=%08x, handle=%p, tr=%02x, blend=%04x, {%d,%d,%d,%d}, {%d,%d,%d,%d}",
-            l->compositionType, l->flags, l->handle, l->transform, l->blending,
-            l->sourceCrop.left,
-            l->sourceCrop.top,
-            l->sourceCrop.right,
-            l->sourceCrop.bottom,
-            l->displayFrame.left,
-            l->displayFrame.top,
-            l->displayFrame.right,
-            l->displayFrame.bottom);
-}
-
-static int hwc_prepare(hwc_composer_device_t *dev, hwc_layer_list_t* list) {
-    if (list && (list->flags & HWC_GEOMETRY_CHANGED)) {
-        for (size_t i=0 ; i<list->numHwLayers ; i++) {
-            //dump_layer(&list->hwLayers[i]);
-            list->hwLayers[i].compositionType = HWC_FRAMEBUFFER;
-        }
-    }
-    return 0;
-}
-
-static int hwc_set(hwc_composer_device_t *dev,
-        hwc_display_t dpy,
-        hwc_surface_t sur,
-        hwc_layer_list_t* list)
-{
-    //for (size_t i=0 ; i<list->numHwLayers ; i++) {
-    //    dump_layer(&list->hwLayers[i]);
-    //}
-
-    EGLBoolean sucess = eglSwapBuffers((EGLDisplay)dpy, (EGLSurface)sur);
-    if (!sucess) {
-        return HWC_EGL_ERROR;
-    }
-    return 0;
-}
-
-static int hwc_device_close(struct hw_device_t *dev)
-{
-    struct hwc_context_t* ctx = (struct hwc_context_t*)dev;
-    if (ctx) {
-        free(ctx);
-    }
-    return 0;
-}
-
-/*****************************************************************************/
-
-static int hwc_device_open(const struct hw_module_t* module, const char* name,
-        struct hw_device_t** device)
-{
-    int status = -EINVAL;
-    if (!strcmp(name, HWC_HARDWARE_COMPOSER)) {
-        struct hwc_context_t *dev;
-        dev = (hwc_context_t*)malloc(sizeof(*dev));
-
-        /* initialize our state here */
-        memset(dev, 0, sizeof(*dev));
-
-        /* initialize the procs */
-        dev->device.common.tag = HARDWARE_DEVICE_TAG;
-        dev->device.common.version = 0;
-        dev->device.common.module = const_cast<hw_module_t*>(module);
-        dev->device.common.close = hwc_device_close;
-
-        dev->device.prepare = hwc_prepare;
-        dev->device.set = hwc_set;
-
-        *device = &dev->device.common;
-        status = 0;
-    }
-    return status;
-}
diff --git a/mx6/hwcomposer/Android.mk b/mx6/hwcomposer/Android.mk
index d36ebbb..9ac30d2 100644
--- a/mx6/hwcomposer/Android.mk
+++ b/mx6/hwcomposer/Android.mk
@@ -18,6 +18,7 @@ LOCAL_PATH := $(call my-dir)
 # HAL module implemenation, not prelinked and stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
 ifeq ($(TARGET_HAVE_IMX_HWCOMPOSER),true)
+ifeq ($(HAVE_FSL_IMX_GPU2D),true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -53,3 +54,4 @@ endif
 LOCAL_MODULE_TAGS := optional
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
diff --git a/mx6/libcamera/Android.mk b/mx6/libcamera/Android.mk
index be78565..30403ea 100644
--- a/mx6/libcamera/Android.mk
+++ b/mx6/libcamera/Android.mk
@@ -15,7 +15,7 @@
 LOCAL_PATH:= $(call my-dir)
 
 ifeq ($(BOARD_HAVE_IMX_CAMERA),true)
-
+ifneq ($(IMX_CAMERA_HAL_V2),true)
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES:=    \
@@ -100,4 +100,5 @@ LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
 
diff --git a/mx6/libcamera2/Android.mk b/mx6/libcamera2/Android.mk
index 1f11dd7..0806bf9 100644
--- a/mx6/libcamera2/Android.mk
+++ b/mx6/libcamera2/Android.mk
@@ -15,6 +15,7 @@
 LOCAL_PATH:= $(call my-dir)
 
 ifeq ($(BOARD_HAVE_IMX_CAMERA),true)
+ifeq ($(IMX_CAMERA_HAL_V2),true)
 
 include $(CLEAR_VARS)
 
@@ -98,4 +99,5 @@ LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
 
diff --git a/mx6/libgralloc_wrapper/Android.mk b/mx6/libgralloc_wrapper/Android.mk
index c9f4e81..b80606c 100755
--- a/mx6/libgralloc_wrapper/Android.mk
+++ b/mx6/libgralloc_wrapper/Android.mk
@@ -18,6 +18,7 @@ LOCAL_PATH := $(call my-dir)
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
 
 ifeq ($(TARGET_HAVE_IMX_GRALLOC),true)
+ifeq ($(HAVE_FSL_IMX_GPU2D),true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -59,5 +60,6 @@ LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
 
 
diff --git a/mx7/gralloc/Android.mk b/mx7/gralloc/Android.mk
index 01dfee3..126fa2b 100644
--- a/mx7/gralloc/Android.mk
+++ b/mx7/gralloc/Android.mk
@@ -19,6 +19,7 @@ LOCAL_PATH := $(call my-dir)
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
 
 ifeq ($(TARGET_HAVE_IMX_GRALLOC),true)
+ifneq ($(HAVE_FSL_IMX_GPU2D),true)
 include $(CLEAR_VARS)
 
 LOCAL_MODULE_RELATIVE_PATH := hw
@@ -37,3 +38,4 @@ endif
 
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
diff --git a/mx7/hwcomposer/Android.mk b/mx7/hwcomposer/Android.mk
index 43c94fa..983b896 100644
--- a/mx7/hwcomposer/Android.mk
+++ b/mx7/hwcomposer/Android.mk
@@ -18,6 +18,7 @@ LOCAL_PATH := $(call my-dir)
 # HAL module implemenation, not prelinked and stored in
 # hw/<OVERLAY_HARDWARE_MODULE_ID>.<ro.product.board>.so
 ifeq ($(TARGET_HAVE_IMX_HWCOMPOSER),true)
+ifneq ($(HAVE_FSL_IMX_GPU2D),true)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
@@ -41,3 +42,4 @@ LOCAL_CFLAGS:= -DLOG_TAG=\"hwcomposer\"
 LOCAL_MODULE_TAGS := optional
 include $(BUILD_SHARED_LIBRARY)
 endif
+endif
-- 
1.8.0

